### ISSUE TITLE
New Controller - Ikea STYRBAR E2001/E2002

### ISSUE THREAD

**Issue by arbigator** [EPMatt/awesome-ha-blueprints#94]
Thank you for taking the time to report here! :tada:

**Please don't delete any part of the template**, since keeping the provided structure will help maintainers to process your request more rapidly.

This template should only be used to propose support for a a new Controller blueprint. If this is not your case, you can [choose a different issue template](https://github.com/EPMatt/awesome-ha-blueprints/issues/new/choose). If you have a question instead, please open a discussion [here](https://github.com/EPMatt/awesome-ha-blueprints/discussions/new).

\*_=required_

## Controller Information\*

- Manufacturer: IKEA
- Model: E2002
- Short Description: STYRBAR 4 Button Remote
- Image: 
<img width="173" alt="styrbar" src="https://user-images.githubusercontent.com/85900338/123156516-9fd03400-d426-11eb-9572-8eec31aabbbb.png">

## Actions

If possible, provide here mappings for actions which can be triggered with this controller.
Specify the integration these mappings refer to. If you want to provide mappings for multiple integrations, you can duplicate the following section as needed.

### [Integration] ZHA

- `action_name (e.g. button_press)`: action description (e.g. sent when pressing the button)
- etc.

## Additional Details\*

Similar function to the Tradfri 5 button remote, except short press of big sun turns light on, short press of little sun turns light off.  Long presses dim up/down.

- [ ] I'd like to help developing this blueprint
- [ ] I've found a blueprint/automation not included in this project which integrates this controller
  - Link to the resources you've found: [Resource name here](https://link-goes-here)

## Additional context

Add any other context about the controller here (pitfalls in the device, checks which should be implemented to provide a fully working integration for the device, etc.).

**Comment by EPMatt** [comment]
Hi @arbigator, thank you for taking the time to report!

I was able to find action mappings both for Zigbee2MQTT and deCONZ here: 
https://xaviml.github.io/controllerx/controllers/W2049

Could you please help identifying actions for ZHA?

You can navigate to _Developer Tools_ -> _Events_ in your Home Assistant UI, then in the _Listen to events_ form on the bottom of the page  type `zha_event` and click on _Start Listening_.

When interacting with the controller, you should see a list of events appearing on the Home Assistant UI. Copy the events you received and paste them here, making sure to clearly separate each action/subaction combination, so it's easier to identify which events are fired when a certain action is taken on the device. Repeat the operation for each controller button, and for each button subaction (ie. short press, long press, release after long press).

We can then proceed to develop the blueprint for this controller and add it to the collection.

Thank you for your help!

**Comment by arbigator** [comment]
Thanks for your reply!

Listed below are all the events from the controller. I noticed that every button press creates two events, and in the case of a long press, 4 events. Is this normal? 

There does not seem to be any event triggered for a short press of the left/right buttons.  Events are only triggered from long presses, seems odd to me.

I'm super new to all of this, so I hope this helps and is laid out in an easy to read way. Let me know if there's any more info you need. 

Thanks for al the hard work you've put in to these blueprints!!

Top button short press:
>Event 1 fired 5:15 PM:
{
    "event_type": "zha_event",
    "data": {
        "device_ieee": "84:2e:14:ff:fe:45:55:35",
        "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0006",
        "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
        "endpoint_id": 1,
        "cluster_id": 6,
        "command": "on",
        "args": []
    },
    "origin": "LOCAL",
    "time_fired": "2021-07-06T23:15:32.931723+00:00",
    "context": {
        "id": "a9d20a7ef1d50ba4906cb9bb15f8cc11",
        "parent_id": null,
        "user_id": null
    }
}
Event 0 fired 5:15 PM:
{
    "event_type": "zha_event",
    "data": {
        "device_ieee": "84:2e:14:ff:fe:45:55:35",
        "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0006",
        "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
        "endpoint_id": 1,
        "cluster_id": 6,
        "command": "on",
        "args": []
    },
    "origin": "LOCAL",
    "time_fired": "2021-07-06T23:15:32.909633+00:00",
    "context": {
        "id": "e595905a920bbfa0af2b9f5505241995",
        "parent_id": null,
        "user_id": null
    }
}

Top button long press:
>Event 5 fired 5:18 PM:
{
    "event_type": "zha_event",
    "data": {
        "device_ieee": "84:2e:14:ff:fe:45:55:35",
        "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0008",
        "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
        "endpoint_id": 1,
        "cluster_id": 8,
        "command": "stop",
        "args": []
    },
    "origin": "LOCAL",
    "time_fired": "2021-07-06T23:18:57.693848+00:00",
    "context": {
        "id": "4cc3a3a0e5512b0e07d2d1a449ec190d",
        "parent_id": null,
        "user_id": null
    }
}
Event 4 fired 5:18 PM:
{
    "event_type": "zha_event",
    "data": {
        "device_ieee": "84:2e:14:ff:fe:45:55:35",
        "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0008",
        "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
        "endpoint_id": 1,
        "cluster_id": 8,
        "command": "stop",
        "args": []
    },
    "origin": "LOCAL",
    "time_fired": "2021-07-06T23:18:57.673793+00:00",
    "context": {
        "id": "4b41cc324c36bd4b5a17ddf59dd04c26",
        "parent_id": null,
        "user_id": null
    }
}
Event 3 fired 5:18 PM:
{
    "event_type": "zha_event",
    "data": {
        "device_ieee": "84:2e:14:ff:fe:45:55:35",
        "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0008",
        "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
        "endpoint_id": 1,
        "cluster_id": 8,
        "command": "move_with_on_off",
        "args": [
            0,
            83
        ]
    },
    "origin": "LOCAL",
    "time_fired": "2021-07-06T23:18:57.246069+00:00",
    "context": {
        "id": "788601a27a24587867f75f5172afedf3",
        "parent_id": null,
        "user_id": null
    }
}
Event 2 fired 5:18 PM:
{
    "event_type": "zha_event",
    "data": {
        "device_ieee": "84:2e:14:ff:fe:45:55:35",
        "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0008",
        "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
        "endpoint_id": 1,
        "cluster_id": 8,
        "command": "move_with_on_off",
        "args": [
            0,
            83
        ]
    },
    "origin": "LOCAL",
    "time_fired": "2021-07-06T23:18:57.185849+00:00",
    "context": {
        "id": "69f81a14977ad3a702bbafec7b2d3c47",
        "parent_id": null,
        "user_id": null
    }
}

Bottom button short press
> {
    "event_type": "zha_event",
    "data": {
        "device_ieee": "84:2e:14:ff:fe:45:55:35",
        "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0006",
        "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
        "endpoint_id": 1,
        "cluster_id": 6,
        "command": "off",
        "args": []
    },
    "origin": "LOCAL",
    "time_fired": "2021-07-06T23:51:12.975685+00:00",
    "context": {
        "id": "11aeb178ddfcecc4a905659e1507ba67",
        "parent_id": null,
        "user_id": null
    }
}
Event 12 fired 5:51 PM:
{
    "event_type": "zha_event",
    "data": {
        "device_ieee": "84:2e:14:ff:fe:45:55:35",
        "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0006",
        "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
        "endpoint_id": 1,
        "cluster_id": 6,
        "command": "off",
        "args": []
    },
    "origin": "LOCAL",
    "time_fired": "2021-07-06T23:51:12.941442+00:00",
    "context": {
        "id": "48d2100bbfe6543be32fb9948af389d6",
        "parent_id": null,
        "user_id": null
    }
}

Bottom button long press
>Event 17 fired 5:51 PM:
{
    "event_type": "zha_event",
    "data": {
        "device_ieee": "84:2e:14:ff:fe:45:55:35",
        "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0008",
        "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
        "endpoint_id": 1,
        "cluster_id": 8,
        "command": "stop",
        "args": []
    },
    "origin": "LOCAL",
    "time_fired": "2021-07-06T23:51:54.487794+00:00",
    "context": {
        "id": "e09109f4266774f3359206f47a450245",
        "parent_id": null,
        "user_id": null
    }
}
Event 16 fired 5:51 PM:
{
    "event_type": "zha_event",
    "data": {
        "device_ieee": "84:2e:14:ff:fe:45:55:35",
        "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0008",
        "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
        "endpoint_id": 1,
        "cluster_id": 8,
        "command": "stop",
        "args": []
    },
    "origin": "LOCAL",
    "time_fired": "2021-07-06T23:51:54.458663+00:00",
    "context": {
        "id": "c0517144c6359207647a0d7335e2d99a",
        "parent_id": null,
        "user_id": null
    }
}
Event 15 fired 5:51 PM:
{
    "event_type": "zha_event",
    "data": {
        "device_ieee": "84:2e:14:ff:fe:45:55:35",
        "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0008",
        "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
        "endpoint_id": 1,
        "cluster_id": 8,
        "command": "move",
        "args": [
            1,
            83
        ]
    },
    "origin": "LOCAL",
    "time_fired": "2021-07-06T23:51:51.052925+00:00",
    "context": {
        "id": "f21933cce04d80fb1b82bcd6759a8588",
        "parent_id": null,
        "user_id": null
    }
}
Event 14 fired 5:51 PM:
{
    "event_type": "zha_event",
    "data": {
        "device_ieee": "84:2e:14:ff:fe:45:55:35",
        "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0008",
        "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
        "endpoint_id": 1,
        "cluster_id": 8,
        "command": "move",
        "args": [
            1,
            83
        ]
    },
    "origin": "LOCAL",
    "time_fired": "2021-07-06T23:51:51.026169+00:00",
    "context": {
        "id": "e769d5f7c4cc64d5b599b1b8bfab22c5",
        "parent_id": null,
        "user_id": null
    }
}

Right arrow long press:
>Event 19 fired 5:55 PM:
{
    "event_type": "zha_event",
    "data": {
        "device_ieee": "84:2e:14:ff:fe:45:55:35",
        "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0006",
        "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
        "endpoint_id": 1,
        "cluster_id": 6,
        "command": "on",
        "args": []
    },
    "origin": "LOCAL",
    "time_fired": "2021-07-06T23:55:10.383000+00:00",
    "context": {
        "id": "6170becf9395fed97b30cb374fc33d6c",
        "parent_id": null,
        "user_id": null
    }
}
Event 18 fired 5:55 PM:
{
    "event_type": "zha_event",
    "data": {
        "device_ieee": "84:2e:14:ff:fe:45:55:35",
        "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0006",
        "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
        "endpoint_id": 1,
        "cluster_id": 6,
        "command": "on",
        "args": []
    },
    "origin": "LOCAL",
    "time_fired": "2021-07-06T23:55:10.359633+00:00",
    "context": {
        "id": "885ed9f27438b31c9fdfdbc06c3b3863",
        "parent_id": null,
        "user_id": null
    }
}

Left arrow long press:
>Event 21 fired 5:56 PM:
{
    "event_type": "zha_event",
    "data": {
        "device_ieee": "84:2e:14:ff:fe:45:55:35",
        "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0006",
        "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
        "endpoint_id": 1,
        "cluster_id": 6,
        "command": "on",
        "args": []
    },
    "origin": "LOCAL",
    "time_fired": "2021-07-06T23:56:35.008056+00:00",
    "context": {
        "id": "eff76826bd39035b462bbb5a962c76f8",
        "parent_id": null,
        "user_id": null
    }
}
Event 20 fired 5:56 PM:
{
    "event_type": "zha_event",
    "data": {
        "device_ieee": "84:2e:14:ff:fe:45:55:35",
        "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0006",
        "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
        "endpoint_id": 1,
        "cluster_id": 6,
        "command": "on",
        "args": []
    },
    "origin": "LOCAL",
    "time_fired": "2021-07-06T23:56:34.973410+00:00",
    "context": {
        "id": "cc3f5b08d27ba9f1f8dd0d3326649472",
        "parent_id": null,
        "user_id": null
    }
}

**Comment by EPMatt** [comment]
Hi @abrigator, sorry for my late reply.

Thank you so much for your feedback. 

>  I noticed that every button press creates two events, and in the case of a long press, 4 events. Is this normal?

Yeah, many controllers report more events that what's needed (especially Philips controllers in my very little experience), but luckily our blueprints do not fear duplicate events thanks to the debouncing feature, which we should suggest to enable for this controller. :) 

From the zha events you reported here it's not possible to distinguish a left/right long press from an up button short press, since the main event data are exactly the same: 
```json
"data": {
"device_ieee": "84:2e:14:ff:fe:45:55:35",
"unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0006",
"device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
"endpoint_id": 1,
"cluster_id": 6,
"command": "on",
"args": []
},
```

Also, it's strange that no events are fired for short presses both on the left and right buttons. This is probably an issue with the device integration in ZHA, hence I'd suggest you to open an issue on the Home Assistant Core repo, referencing this issue so that we can keep track of the progress here.

In the meanwhile, I'll open a draft PR and start working on the blueprint for this device. Whenever the issue with left/right buttons events is addressed, we can finalize the blueprint and make it available to the public.

Thank you for your help!

**Comment by arbigator** [comment]
Hi @EPMatt,

Sorry for the huge delay, I basically forgot I had this controller, but picked it up again the other day. I re-paired it to Home Assistant and got it working a little better than before. I removed all of my devices in ZHA and started fresh to get it to pair properly. In doing this, I was able to get it to send different commands than I had gotten before. I've included them below for you. There's still some weirdness for long presses of the left and right button, in total it sends about 15 commands for a long press+release, but only 5 different commands repeated 3x each. 

Top Button Short Press

 

> {
>     "event_type": "zha_event",
>     "data": {
>         "device_ieee": "84:2e:14:ff:fe:45:55:35",
>         "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0006",
>         "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
>         "endpoint_id": 1,
>         "cluster_id": 6,
>         "command": "on",
>         "args": []
>     },
>     "origin": "LOCAL",
>     "time_fired": "2021-10-02T17:40:27.464695+00:00",
>     "context": {
>         "id": "ae2e0a00f5fb65789279c9934ff2ee7a",
>         "parent_id": null,
>         "user_id": null
>     }
> }


Top Button Long Press

> {
>     "event_type": "zha_event",
>     "data": {
>         "device_ieee": "84:2e:14:ff:fe:45:55:35",
>         "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0008",
>         "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
>         "endpoint_id": 1,
>         "cluster_id": 8,
>         "command": "move_with_on_off",
>         "args": [
>             0,
>             83
>         ]
>     },
>     "origin": "LOCAL",
>     "time_fired": "2021-10-02T17:43:25.986460+00:00",
>     "context": {
>         "id": "4469f3de255aca0e1ef9b8d3b134c905",
>         "parent_id": null,
>         "user_id": null
>     }
> }

Top Button Release After Long Press

> {
>     "event_type": "zha_event",
>     "data": {
>         "device_ieee": "84:2e:14:ff:fe:45:55:35",
>         "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0008",
>         "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
>         "endpoint_id": 1,
>         "cluster_id": 8,
>         "command": "stop",
>         "args": []
>     },
>     "origin": "LOCAL",
>     "time_fired": "2021-10-02T17:43:28.205544+00:00",
>     "context": {
>         "id": "49c43671c0f0f7d8c7a5882a646a6c3d",
>         "parent_id": null,
>         "user_id": null
>     }
> }

Bottom Button Short Press

> {
>     "event_type": "zha_event",
>     "data": {
>         "device_ieee": "84:2e:14:ff:fe:45:55:35",
>         "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0006",
>         "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
>         "endpoint_id": 1,
>         "cluster_id": 6,
>         "command": "off",
>         "args": []
>     },
>     "origin": "LOCAL",
>     "time_fired": "2021-10-02T17:42:03.506815+00:00",
>     "context": {
>         "id": "73c9b843dbb19986d4e5cd2a880bcbc6",
>         "parent_id": null,
>         "user_id": null
>     }
> }

Bottom Button Long Press

> {
>     "event_type": "zha_event",
>     "data": {
>         "device_ieee": "84:2e:14:ff:fe:45:55:35",
>         "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0008",
>         "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
>         "endpoint_id": 1,
>         "cluster_id": 8,
>         "command": "move",
>         "args": [
>             1,
>             83
>         ]
>     },
>     "origin": "LOCAL",
>     "time_fired": "2021-10-02T17:55:47.281531+00:00",
>     "context": {
>         "id": "eafdbdf083aee2e4b575ee58dc96a1de",
>         "parent_id": null,
>         "user_id": null
>     }
> }

Bottom Button Release After Long Press

> {
>     "event_type": "zha_event",
>     "data": {
>         "device_ieee": "84:2e:14:ff:fe:45:55:35",
>         "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0008",
>         "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
>         "endpoint_id": 1,
>         "cluster_id": 8,
>         "command": "stop",
>         "args": []
>     },
>     "origin": "LOCAL",
>     "time_fired": "2021-10-02T17:55:49.084689+00:00",
>     "context": {
>         "id": "faf19c7777e68e7f4882e35bc491f3e4",
>         "parent_id": null,
>         "user_id": null
>     }
> }

Right Button Short Press

> {
>     "event_type": "zha_event",
>     "data": {
>         "device_ieee": "84:2e:14:ff:fe:45:55:35",
>         "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0005",
>         "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
>         "endpoint_id": 1,
>         "cluster_id": 5,
>         "command": "press",
>         "args": [
>             256,
>             13,
>             0
>         ]
>     },
>     "origin": "LOCAL",
>     "time_fired": "2021-10-02T17:41:40.658739+00:00",
>     "context": {
>         "id": "e4ee5602b8560224962c4b223a754eff",
>         "parent_id": null,
>         "user_id": null
>     }
> }

Left Button Short Press

> {
>     "event_type": "zha_event",
>     "data": {
>         "device_ieee": "84:2e:14:ff:fe:45:55:35",
>         "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0005",
>         "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
>         "endpoint_id": 1,
>         "cluster_id": 5,
>         "command": "press",
>         "args": [
>             257,
>             13,
>             0
>         ]
>     },
>     "origin": "LOCAL",
>     "time_fired": "2021-10-02T17:42:26.226842+00:00",
>     "context": {
>         "id": "2bdd612c940e6d27a0e376b5b16cd335",
>         "parent_id": null,
>         "user_id": null
>     }
> }



Long press and hold+release of the right button sends the following 5 commands. After some time, it sends the "hold" command.

> Event 110 fired 11:46 AM:
> {
>     "event_type": "zha_event",
>     "data": {
>         "device_ieee": "84:2e:14:ff:fe:45:55:35",
>         "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0005",
>         "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
>         "endpoint_id": 1,
>         "cluster_id": 5,
>         "command": "release",
>         "args": [
>             -27903
>         ]
>     },
>     "origin": "LOCAL",
>     "time_fired": "2021-10-02T17:46:39.445960+00:00",
>     "context": {
>         "id": "b095d4076a16933213677d094c1a933b",
>         "parent_id": null,
>         "user_id": null
>     }
> }
> 
> Event 107 fired 11:46 AM:
> {
>     "event_type": "zha_event",
>     "data": {
>         "device_ieee": "84:2e:14:ff:fe:45:55:35",
>         "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0005",
>         "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
>         "endpoint_id": 1,
>         "cluster_id": 5,
>         "command": "hold",
>         "args": [
>             3328,
>             0
>         ]
>     },
>     "origin": "LOCAL",
>     "time_fired": "2021-10-02T17:46:37.769353+00:00",
>     "context": {
>         "id": "218b7e1579f4b71ee1945450d04cad20",
>         "parent_id": null,
>         "user_id": null
>     }
> }
> 
> Event 104 fired 11:46 AM:
> {
>     "event_type": "zha_event",
>     "data": {
>         "device_ieee": "84:2e:14:ff:fe:45:55:35",
>         "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0005",
>         "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
>         "endpoint_id": 1,
>         "cluster_id": 5,
>         "command": "press",
>         "args": [
>             2,
>             0,
>             0
>         ]
>     },
>     "origin": "LOCAL",
>     "time_fired": "2021-10-02T17:46:36.773098+00:00",
>     "context": {
>         "id": "4fdbc1bebecc4c8939f3701e9e4630bb",
>         "parent_id": null,
>         "user_id": null
>     }
> }
> Event 101 fired 11:46 AM:
> {
>     "event_type": "zha_event",
>     "data": {
>         "device_ieee": "84:2e:14:ff:fe:45:55:35",
>         "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0006",
>         "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
>         "endpoint_id": 1,
>         "cluster_id": 6,
>         "command": "on",
>         "args": []
>     },
>     "origin": "LOCAL",
>     "time_fired": "2021-10-02T17:46:36.274636+00:00",
>     "context": {
>         "id": "87fef9dacac827604c24f16ea3f0288a",
>         "parent_id": null,
>         "user_id": null
>     }
> }
> 
> Event 98 fired 11:46 AM:
> {
>     "event_type": "zha_event",
>     "data": {
>         "device_ieee": "84:2e:14:ff:fe:45:55:35",
>         "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0005",
>         "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
>         "endpoint_id": 1,
>         "cluster_id": 5,
>         "command": "release",
>         "args": [
>             0
>         ]
>     },
>     "origin": "LOCAL",
>     "time_fired": "2021-10-02T17:46:35.765670+00:00",
>     "context": {
>         "id": "759917662d68c96b3cea5216cb9fe136",
>         "parent_id": null,
>         "user_id": null
>     }
> }


Long press and hold+release of the left button sends the following 5 commands. After some time, it sends the "hold" command.

> Event 404 fired 11:59 AM:
> {
>     "event_type": "zha_event",
>     "data": {
>         "device_ieee": "84:2e:14:ff:fe:45:55:35",
>         "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0005",
>         "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
>         "endpoint_id": 1,
>         "cluster_id": 5,
>         "command": "release",
>         "args": [
>             1365
>         ]
>     },
>     "origin": "LOCAL",
>     "time_fired": "2021-10-02T17:59:09.481005+00:00",
>     "context": {
>         "id": "52e87a5cad0dfecaf04c31419385232e",
>         "parent_id": null,
>         "user_id": null
>     }
> }
> 
> Event 401 fired 11:59 AM:
> {
>     "event_type": "zha_event",
>     "data": {
>         "device_ieee": "84:2e:14:ff:fe:45:55:35",
>         "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0005",
>         "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
>         "endpoint_id": 1,
>         "cluster_id": 5,
>         "command": "hold",
>         "args": [
>             3329,
>             0
>         ]
>     },
>     "origin": "LOCAL",
>     "time_fired": "2021-10-02T17:59:08.128209+00:00",
>     "context": {
>         "id": "07977accca88b6fa0a42788d634a144f",
>         "parent_id": null,
>         "user_id": null
>     }
> }
> 
> Event 398 fired 11:59 AM:
> {
>     "event_type": "zha_event",
>     "data": {
>         "device_ieee": "84:2e:14:ff:fe:45:55:35",
>         "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0005",
>         "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
>         "endpoint_id": 1,
>         "cluster_id": 5,
>         "command": "press",
>         "args": [
>             2,
>             0,
>             0
>         ]
>     },
>     "origin": "LOCAL",
>     "time_fired": "2021-10-02T17:59:07.125595+00:00",
>     "context": {
>         "id": "a5421995e5693ef50b486ea45432098b",
>         "parent_id": null,
>         "user_id": null
>     }
> }
> 
> Event 395 fired 11:59 AM:
> {
>     "event_type": "zha_event",
>     "data": {
>         "device_ieee": "84:2e:14:ff:fe:45:55:35",
>         "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0006",
>         "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
>         "endpoint_id": 1,
>         "cluster_id": 6,
>         "command": "on",
>         "args": []
>     },
>     "origin": "LOCAL",
>     "time_fired": "2021-10-02T17:59:06.641533+00:00",
>     "context": {
>         "id": "b02c86d91f3668c9e3b83fc2ff7bda4c",
>         "parent_id": null,
>         "user_id": null
>     }
> }
> 
> Event 392 fired 11:59 AM:
> {
>     "event_type": "zha_event",
>     "data": {
>         "device_ieee": "84:2e:14:ff:fe:45:55:35",
>         "unique_id": "84:2e:14:ff:fe:45:55:35:1:0x0005",
>         "device_id": "eb86c87f4ddd65245b1ca25e2be72a71",
>         "endpoint_id": 1,
>         "cluster_id": 5,
>         "command": "release",
>         "args": [
>             0
>         ]
>     },
>     "origin": "LOCAL",
>     "time_fired": "2021-10-02T17:59:06.127411+00:00",
>     "context": {
>         "id": "62f907c081f7a6ca723aa89fbc2f7057",
>         "parent_id": null,
>         "user_id": null
>     }
> }

Thanks for your time, and apologies again for forgetting about this. Hopefully this is helpful to you in some way. Let me know if you need anything else!

**Comment by EPMatt** [comment]
Hi @arbigator,

I'm sorry for the delay. Thank you so much for your precious feedback. With the info in your latest comment I was able to properly add ZHA mappings for this controller. :) 
I've just opened a PR (#171) which includes the controller blueprint for the IKEA E2001/E2002, as well as adding support for the Light and Media Player hooks. 

We can continue the discussion regarding this feature in the linked PR.

Thank you so much for your help and your patience. :)
